src/controller/user/Controller.ts

```diff
@@ -3,168 +3,196 @@ import UserRepository from '../../repositories/user/UserRepository';
+			}
+			const updateUser = await userRepository.update(userData);
+			return res.status(200).send({ message: 'user updated successfully', UserData: updateUser });
+		} catch (error) {
+			return res.status(500).send({ err: 'Server Error', message: 'Something went wrong' });
+		}
+	}
 
-  /**
-   * @description Create Token BY ID and Email
-   * @param req
-   * @param res
-   * @param next
-   * @returns New JWT Token
-   */
-  public async createToken(req: Request, res: Response, next: NextFunction) {
-    try {
-      const { id, email } = req.body;
-      const token = await jwt.sign(req.body, config.secret, { expiresIn: '10h' });
-      return res.status(200).send({
-        message: 'token successfully created',
-        data: { token },
-        status: 200,
-      });
-    } catch (error) {
-      return res.status(500).send({ err: 'Server Error', message: 'Something went wrong' });
-    }
-  }
+	/**
+	 * @description Delete User Data by respected ID
+	 * @param req
+	 * @param res
+	 * @param next
+	 * @returns SoftDeleted Data
+	 */
+	public async delete(req: Request, res: Response, next: NextFunction) {
+		try {
+			const Id = req.params.id;
+			const findUser = await userRepository.findOneData({ _id: Id });
+			console.log('findFor delete in User controller', { findUser });
+			const userData = await userRepository.delete(findUser);
+			console.log('deleteData in User controller', { userData });
+			return res.status(200).send({
+				message: 'user deleted successfully',
+				deleted_User: userData,
+			});
+		} catch (error) {
+			return res.status(500).send({ err: 'server error', message: 'Something went Wrong' });
+		}
+	}
+
+	/**
+	 * @description Create Token BY ID and Email
+	 * @param req
+	 * @param res
+	 * @param next
+	 * @returns New JWT Token
+	 */
+	public async createToken(req: Request, res: Response, next: NextFunction) {
+		try {
+			// const { id, email } = req.body;
+			const token = await jwt.sign(req.body, config.secret, { expiresIn: '15m' });
+			return res.status(200).send({
+				message: 'token successfully created',
+				data: { token },
+				status: 200,
+			});
+		} catch (error) {
+			return res.status(500).send({ err: 'Server Error', message: 'Something went wrong' });
+		}
+	}
+	/**
+	 * Login
+	 * @param req
+	 * @param res
+	 * @param next
+	 * @returns Token
+	 */
+	public async login(req: Request, res: Response, next: NextFunction) {
+		try {
+			const { email, password } = req.body;
+			const user = await userRepository.findOneData({ deletedAt: null, email: email });
+			if (!user) {
+				return next({ status: 404, error: 'bad request', message: 'User Not found' });
+			}
+			const validPassword = bcrypt.compareSync(password, user.password);
+      // console.log({validPassword});
+      
+			if (!validPassword) {
+				return next({ success: false, message: 'Password not matched' });
+			}
+			const token = await jwt.sign(req.body, config.secret, { expiresIn: '15m' });
```

Comment:
req.body would not sent.
Either user or some other sepcific value would be sent in token

Reply:
fixed as per feedback